python/hidet/graph/ops/normalize/norm.py

```diff
@@ -352,6 +347,105 @@ def norm_kernel(x: dtype[x.shape], y: dtype[y.shape]):
+                            M2_vec = avx_f32x8_add(M2_vec, avx_f32x8_multiply(delta, delta2))
+
+                        # welford combine
+                        # TODO: case for numerical stability? (number too high for large matrix)
+                        # TODO: look at the cascade thing in pytorch github
+                        mean_combined = avx_f32x8_find_sum(mean_vec) / 8
+                        mean_combined_vec = avx_f32x8_set1(mean_combined)
+                        delta_vec = avx_f32x8_subtract(mean_vec, mean_combined_vec)
+                        M2_combined = avx_f32x8_find_sum(M2_vec) + avx_f32x8_find_sum(
+                            avx_f32x8_multiply(delta_vec, delta_vec)) * (tail_size // 8)
+                    mean_tail = 0.0
+                    M2_tail = 0.0
+                    # welford on remaining parts past 8
+                    for i in range(tail_size % 8):
+                        tail_idx = spatial(*tail).map(tail_size - tail_size % 8 + i)
+                        delta_tail = x[head_idx][tail_idx] - mean_tail
+                        mean_tail += delta_tail / cast(i+1, float32)
+                        delta_tail2 = x[head_idx][tail_idx] - mean_tail
+                        M2_tail += delta_tail * delta_tail2
+                    # welford combine vectorized and unvectorized
+                    delta_end = mean_tail - mean_combined
+                    mean = (mean_combined * (tail_size - tail_size % 8) + mean_tail * (tail_size % 8)) / tail_size
+                    var = (M2_combined + M2_tail + delta_end * delta_end * (tail_size - tail_size % 8) * (tail_size % 8)
+                           / tail_size) / tail_size
+                    mean_vec = avx_f32x8_set1(mean)
+                    var_vec = avx_f32x8_set1(var)
+                    if tail_size >= 8:
+                        for i in range(tail_size // 8):
+                            # norm calculation
+                            tail_idx = spatial(*tail).map(i * 8)
+                            temp_out = tensor(dtype=float32, shape=[8])
+                            avx_f32x8_store(temp_out,
+                                            avx_f32x8_divide(avx_f32x8_subtract(avx_f32x8_load(
+                                                ~x[head_idx][tail_idx]), mean_vec),
+                                                avx_f32x8_sqrt(avx_f32x8_add(var_vec, epsilon_vec))))
+                            for j in range(8):
+                                tail_idx = spatial(*tail).map(i * 8 + j)
+                                out[head_idx][tail_idx] = temp_out[j]
+                    for i in range(tail_size % 8):
+                        tail_idx = spatial(*tail).map(tail_size - tail_size % 8 + i)
+                        out[head_idx][tail_idx] = \
+                            (x[head_idx][tail_idx] - mean) * prim.rsqrt(var + self.attrs['epsilon'])
+
+        norm_cpu_kernel.kind = "cpu_kernel"
+        avx_f32x8_find_sum.kind = "cpu_internal"
```

Comment:
Better to avoid setting the function attributes outside its definition. Instead, use
```python
from hidet.lang import attrs

@hidet.script
def norm_cpu_kernel(...):
    attrs.func_kind = "cpu_kernel"
    ...
```